cleanlab/experimental/tf_nlp.py

```diff
@@ -0,0 +1,129 @@
+        - `input_ids`: a list or tensorflow tensor of token ids to be fed to the model. Shape = (num_sample, sequence_len).
+        - `attention_mask` a list or tensorflow tensor of indices specifying which tokens should be attended to by the model. Shape = (num_sample, sequence_len).
+
+        sequence_len : int,
+        Tokenized sentences length.
+
+        kwargs : optional,
+        Optional arguments useful to fit a `Tensorflow model`. Refers to Tensorflow documentation for a detailed list of attributes `<https://www.tensorflow.org/api_docs/python/tf/keras/Model#fit>`
+        """
+        super().__init__(**kwargs)
+        self.train_input = train_input
+        self.seq_len = seq_len
+
+    def split_input(self, X):
+        splitted_X = [
+            X[:, : self.seq_len],  # input_ids
+            X[:, self.seq_len :],  # attention_mask
+        ]
+        return splitted_X
+
+    @property
+    def feature_encoder(self):
+        return FunctionTransformer(
+            func=self.split_input,
+        )
+
+    def _get_tf_input(self, ids, train_input):
+        indexed_input_ids = np.array(tf.gather(train_input["input_ids"], indices=ids))
+
+        indexed_attention_mask = np.array(tf.gather(train_input["attention_mask"], indices=ids))
+
+        return np.hstack([indexed_input_ids, indexed_attention_mask])
+
+    def fit(self, ids, y, sample_weight=None, **kwargs):
+        """
+        Constructs and fit a new model using the given data.
+        Parameters
+        ----------
+        ids : array-like of shape (n_samples,)
+            Ids of training samples to be used to train the model.
+        y : array-like of shape (n_samples,)
+            True labels.
+        sample_weight : array-like of shape (n_samples,), default=None
+            Array of weights that are assigned to individual samples.
+            If not provided, then each sample is given unit weight.
+        **kwargs : Dict[str, Any]
+            Extra arguments to route to Tensorflow `Model.fit`.
+        """
+        X = self._get_tf_input(ids, self.train_input)
+        return super().fit(X, y, sample_weight=sample_weight, **kwargs)
+
+    def predict_proba(self, ids, **kwargs):
+        """
+        Returns class probability estimates for the given data.
+        Parameters
+        ----------
+        ids : array-like of shape (n_samples,)
+            Ids of training samples to be used to train the model.
+        **kwargs : Dict[str, Any]
+            Extra arguments to route to Tensorflow ``Model.predict``.
+        """
+
+        X = self._get_tf_input(ids, self.train_input)
+        return super().predict_proba(X, **kwargs)
+
+    def predict(self, ids, **kwargs):
+        """
+        Returns predictions for the given data.
+        Parameters
+        ----------
+        ids : array-like of shape (n_samples,)
+            Ids of training samples to be used to train the model.
```

Comment:
"Ids of training samples to be used to train the model."
->
"Indices of training examples for which to produce predictions."